pandas/io/packers.py

```diff
@@ -595,11 +595,14 @@ def decode(obj):
         pd_dtype = pandas_dtype(dtype)
         np_dtype = pandas_dtype(dtype).base
 
+        # see gh-14901
+        klass_dtype = np_dtype if pd_dtype.name != 'category' else pd_dtype
+
         index = obj[u'index']
         result = globals()[obj[u'klass']](unconvert(obj[u'data'], dtype,
                                                     obj[u'compress']),
                                           index=index,
-                                          dtype=np_dtype,
+                                          dtype=klass_dtype,
```

Comment:
1) `datetimetz` is already tested in this `Series`.

2) No, we cannot just pass in `pd_dtype` because it will fail with tz-aware data-types:
~~~python
>>> from pandas import Series, Timestamp, read_msgpack
>>> s = Series([Timestamp('20130102', tz='US/Eastern')] * 5)
>>> s2 = read_msgpack(s.to_msgpack())
>>>
>>> s1.get_values()
array(['2013-01-02T05:00:00.000000000', '2013-01-02T05:00:00.000000000',
       '2013-01-02T05:00:00.000000000', '2013-01-02T05:00:00.000000000',
       '2013-01-02T05:00:00.000000000'], dtype='datetime64[ns]')
>>>
>>> s2.get_values()
array(['2013-01-02T10:00:00.000000000', '2013-01-02T10:00:00.000000000',
       '2013-01-02T10:00:00.000000000', '2013-01-02T10:00:00.000000000',
       '2013-01-02T10:00:00.000000000'], dtype='datetime64[ns]')
~~~

The reason is that the internal block representation of the `Series` is altered, even though the external values are identical.